src/scene/shader-lib/utils.js

```diff
@@ -12,15 +12,18 @@ import { ShaderGenerator } from './programs/shader-generator.js';
  * graphics device.
  * @param {string} vsName - The vertex shader chunk name.
  * @param {string} fsName - The fragment shader chunk name.
- * @param {boolean} [useTransformFeedback] - Whether to use transform feedback. Defaults to false.
+ * @param {object} [shaderDefinitionOptions] - Additional options that will be added to the shader definition.
+ * @param {boolean} [shaderDefinitionOptions.useTransformFeedback] - Whether to use transform feedback. Defaults to false.
+ * @param {string | string[]} [shaderDefinitionOptions.fragmentOutputTypes] - Fragment shader output types, which default to vec4. Passing a string will set the output type for all color attachments. Passing an array will set the output type for each color attachment.
+ * @see ShaderUtils.createDefinition
  * @returns {Shader} The newly created shader.
  */
-function createShader(device, vsName, fsName, useTransformFeedback = false) {
+function createShader(device, vsName, fsName, shaderDefinitionOptions) {
```

Comment:
both functions in this file are part of the public API and we cannot change their signature. Meaning we can add the options at the end, but after the `useTransformFeedback`, which should not be removed.

Alternatively (probably a better option), as `useTransformFeedback` is pretty uncommon, would be to do what you do, but handle a case where shaderDefinitionOptions is a boolean and not an object, allowing backwards compatibility.